src/reducers/spendings.js

```diff
@@ -0,0 +1,63 @@
+const spendings = [
+    {title: "fsa", amount: 525, id: 0, users: Array(2)},
+    {title: "vodka", amount: 100, id: 0, users: Array(2)}
+];
+
+
+export function spendingsReducer(state = [], action) {
+    switch (action.type) {
+
+        case "ON_ADD_SPENDINGS":
+            let newSpendings;
+            if (state.length === 0) {
+                newSpendings = {
+                    title: action.text,
+                    amount: 0,
+                    id: 0
+                }
+            } else {
+                newSpendings = {
+                    title: action.text,
+                    amount: 0,
+                    id: state[state.length - 1].id + 1
+                }
+            }
+            console.log(state, newSpendings);
+            return [
+                ...state,
+                newSpendings
+            ];
+
+        case "ON_REMOVE_SPENDINGS":
+            let confirm = window.confirm(`Do you really want to remove ${action.title}?`);
+            if (confirm) {
+                let newArr = state.slice(0, action.id).concat(state.slice(action.id + 1));
+                localStorage.setItem('spendings', JSON.stringify(newArr));
+                    return [...newArr];
+            }
+            return state;
+
+        case "GET_ARR_OF_USERS_SPENDS":
```

Comment:
all actions should trigger state modifications and it is not clear from the name of this action, what it modifies in the state.

Before we scrap this action, I'd say we spend some time modeling the state structure.
Can you put here in the comments the state your application currently has?

The goal is to build a state that will contain only the absolute minimum of source data, for ex. users, transactions
and will not contain derivative information which could be easily calculated later one, like sums and totals

Just drop your JSON structure of state here in the comments and we will discuss it.

Reply:
```
{
    users: [
        {
            name: "John",
            id: 0,
            totalSpendings: 0,
            groupSpengings: 0,
            userDebt: 0
        },
        {
            name: "Bohdan",
            id: 0,
            totalSpendings: 0,
            groupSpengings: 0,
            userDebt: 0
        },
        {
            name: "Liza",
            id: 0,
            totalSpendings: 0,
            groupSpengings: 0,
            userDebt: 0
        }
               ]
    spendings: [
        {
            title: "Taxi",
            amount: 525, id: 0,
            users: [
                {
                    name: "John",
                    id: 1,
                    amount: "15"
                }
            ]
        },
        {
            title: "Water",
            amount: 100,
            id: 0,
            users: [
                {
                    name: "John",
                    id: 2,
                    amount: "10"
                }
            ]
        }
    ]
}

```

Reply:
Let's update the state to exclude derivative data